static/planttracer.js

```diff
@@ -19,8 +19,15 @@ const UPLOAD_TIMEOUT_SECONDS = 20;
 
 // sounds for buttons
 var SOUNDS = [];
-SOUNDS[DELETE_BUTTON]   = new Audio('static/pop-up-something-160353.mp3');
-SOUNDS[UNDELETE_BUTTON] = new Audio('static/soap-bubbles-pop-96873.mp3');
+if (typeof Audio !== 'undefined') {
+  SOUNDS[DELETE_BUTTON] = new Audio('static/pop-up-something-160353.mp3');
+  SOUNDS[UNDELETE_BUTTON] = new Audio('static/soap-bubbles-pop-96873.mp3');
+} else {
+  // Provide fallbacks or empty mock objects for testing
+  SOUNDS[DELETE_BUTTON] = { play: () => {} };
+  SOUNDS[UNDELETE_BUTTON] = { play: () => {} };
+}
+
 
```

Comment:
Makes sense - you may not have audio support.

Reply:
Yep, I ran into this issue when writing the tests, and decided that having something there for a fallback couldn't hurt